test/enhancedStepFunctions.js

```diff
@@ -293,8 +293,29 @@ contract('EnhancedStepFunctions', function(accounts) {
         decode = await convRatesInst.mockDecodeStepData(val);
         assert.equal(decode[0].valueOf(), x, "decode x wrong value");
         assert.equal(decode[1].valueOf(), y, "decode y wrong value");
+    });
+
+    it("should return MAX value for X with special value of step X", async function() {
         // checking for overflow
-        x = (new BigNumber(2)).pow(127);
+        let x = (new BigNumber(2)).pow(127).sub(1);
+        let y = 0;
+        let val = await convRatesInst.mockEncodeStepData(x.valueOf(), y);
+        let decode = await convRatesInst.mockDecodeStepData(val);
+        x = await convRatesInst.mockGetImbalanceMax();
+        assert.equal(decode[0].valueOf(), x.valueOf(), "decode x1 wrong value");
```

Comment:
this is a fair solution. for the valueOf problem.
please also add one mock function.
that get as input a uint.
compares to max_imbalance.
if equal return true.
otherwise false.

just to be on the safe side of this weird issue.